drivers/platform/stm32/stm32_pwm.h

```diff
@@ -64,6 +64,8 @@ enum TimOCMode {
 	TIM_OC_PWM2 = 2,
 };
 
+#ifdef HAL_TIM_MODULE_ENABLED
```

Comment:
please remove these from stm32_pwm, it is not necessary. same remark applies as for DMA, why would you use stm32_pwm without a timer module?
there is no HAL_SPI_MODULE_ENABLED check in stm32_spi...